### PR TITLE
Fix an issue when specifying a passphrase that was raising a TypeError

### PR DESCRIPTION
**Pull request by ph**
The passphrase uses the `Password` class to encapsulate the sensitive
information and make sure we dont write it to the log by accident, to
get the original value the plugin must call `#value` on the object.

The Java layer was expecting a string but was receive a Password
instance and was throwing an exception.

This commit make sure we use password and add an integration test for
it.

Fixes: #138

### CODE REVIEW

File: spec/integration/filebeat_spec.rb [hunk 1]
```diff
@@ -33,13 +33,12 @@
...
   let(:filebeat_exec) { FILEBEAT_BINARY }
 
-  let_empty_tmp_file(:registry_file)
   let(:filebeat_config) do
     {
       "filebeat" => {
         "prospectors" => [{ "paths" => [log_file],  "input_type" => "log" }],
-        "registry_file" => registry_file,
-        "scan_frequency" => "1s"
+        "scan_frequency" => "1s",
```
Comment (andrewvc) [inline spec/integration/filebeat_spec.rb]: Why were properties in this file changed for this PR?

File: spec/integration/filebeat_spec.rb [hunk 2]
```diff
@@ -106,8 +105,38 @@
 
       context "self signed certificate" do
         include_examples "send events"
+
+
+        # Refactor this to use Flores's PKI instead of openssl command line
```
Comment (andrewvc) [inline spec/integration/filebeat_spec.rb]: Nice!

File: spec/integration/filebeat_spec.rb [hunk 3]
```diff
@@ -134,10 +163,10 @@
             "output" => {
               "logstash" => {
                 "hosts" => ["#{host}:#{port}"],
-                "tls" => { 
+                "ssl" => {
```
Comment (andrewvc) [inline spec/integration/filebeat_spec.rb]: Why the change from tls to ssl?

**Comment by ph** [comment]
@andrewvc the properties needed to be changed to keep it in sync with what changed from filebeat.

File: spec/integration/filebeat_spec.rb (hunk 1 above)
Comment (ph) [inline spec/integration/filebeat_spec.rb]: `registry_file` doesnt work anymore with FB 5

File: spec/integration/filebeat_spec.rb (hunk 3 above)
Comment (suyograo) [inline spec/integration/filebeat_spec.rb]: Also this is just in the spec

**Comment by andrewvc** [comment]
LGTM
